activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -120,7 +120,7 @@ def create(*arguments)
         $stderr.puts "Database '#{configuration['database']}' already exists"
       rescue Exception => error
         $stderr.puts error
-        $stderr.puts "Couldn't create database for #{configuration.inspect}"
+        $stderr.puts "Couldn't create database for #{configuration.except("username","password").inspect}"
```

Comment:
I was concerned about that, as well.  Like @kaspth says, it is somewhat strange to me that it is logged at all, except in that maybe you can see at a glance that you got a config variable screwed up in the terminal or logs.  

On my fork, I have more work which I can add to this pending more input, wherein I excise the credentials only in the event of a production environment, [here](https://github.com/marrsale/rails/commit/77258a8c5a241c627c10f17186167e113ca7e426).  

Thanks!